Tests/SentryTests/Recording/SentryCrashCTests.swift

```diff
@@ -0,0 +1,282 @@
+@testable import Sentry
+import XCTest
+
+class SentryCrashCTests: XCTestCase {
+    func testOnCrash_notCrashedDuringCrashHandling_shouldWriteReportToDisk() throws {
+        // -- Arrange --
+        var appName = "SentryCrashCTests"
+            .cString(using: .utf8)!
+        let installDir = URL(fileURLWithPath: NSTemporaryDirectory())
+            .appendingPathComponent("SentryCrashCTests-\(UUID().uuidString)")
+        var installPath = installDir
+            .path
+            .cString(using: .utf8)!
+        let expectedReportsDir = installDir
+            .appendingPathComponent("Reports")
+
+        // Smoke test the existence of the directory
+        XCTAssertFalse(FileManager.default.fileExists(
+            atPath: expectedReportsDir.path
+        ))
+
+        // Installing the sentrycrash will setup the exception handler
+        sentrycrash_uninstall()
+        sentrycrash_install(&appName, &installPath)
+
+        var monitorContext = SentryCrash_MonitorContext()
+        monitorContext.crashedDuringCrashHandling = false
+
+        // -- Act --
+        // Calling the handle exception will trigger the onCrash handler
+        sentrycrashcm_handleException(&monitorContext)
+
+        // -- Assert --
+        let reportUrls = try FileManager.default
+            .contentsOfDirectory(atPath: expectedReportsDir.path)
+        XCTAssertEqual(reportUrls.count, 1)
+    }
+
+    func testOnCrash_notCrashedDuringCrashHandling_installFilePathTooLong_shouldNotWriteToDisk() {
```

Comment:
Good on you for encoding behavior in a test like this, but IMO this situation is something we should fix ASAP if it's happening in the wild. We should always expect to be able to deliver crash reports, and guard against situations that would prevent that and work around them whenever possible. At the very least a warning message should be available for end users to also potentially work around it, like if they provide a custom cache path that is too long. The only other thing I can see that would cause this is a bundle id that is too long (we should also only use the bundle id for mac apps; ios/tvos/visionos are all self-contained files trees per app)

Reply:
I had a quick chat with @philipphofmann yesterday, I still need to validate if a too long file path can actually be passed down from the SentrySDK options using `cacheDirectoryPath` to the `SentryCrashC`, or if it causes issues before that.

He also mentioned a warning should be sufficient, but at this time I already created this test case to actually test the behavior with invalid paths.

In general I still believe we should also add tests to pin down expected negative behavior (if possible), so even if an edge case can hardly ever happen, we do have documentation that it can happen. If it then turns out to be an actual issue, we can change test-case, and fix the implementation until it passes again.

Reply:
Definitely 👍🏻